cmd/dryrun/dryrun.go

```diff
@@ -0,0 +1,537 @@
+func (d *DryRunner) readPolicy(cmd *cobra.Command) (*policyv1.ConfigurationPolicy, error) {
+	reader, err := os.Open(d.policyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	policyBytes, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+
+	unstruct := unstructured.Unstructured{}
+
+	if err := k8syaml.UnmarshalStrict(policyBytes, &unstruct.Object); err != nil {
+		return nil, err
+	}
+
+	if unstruct.GetAPIVersion() != "policy.open-cluster-management.io/v1" {
+		return nil, fmt.Errorf("unsupported apiVersion: %v, the input policy must be "+
+			"'policy.open-cluster-management.io/v1'", unstruct.GetAPIVersion())
+	}
+
+	cfgpol := policyv1.ConfigurationPolicy{}
+
+	switch unstruct.GetKind() {
+	case "ConfigurationPolicy":
+		if err := k8syaml.UnmarshalStrict(policyBytes, &cfgpol); err != nil {
+			return nil, fmt.Errorf("could not unmarshal input to a ConfigurationPolicy: %w", err)
+		}
+	case "Policy":
+		tmpls, found, err := unstructured.NestedSlice(unstruct.Object, "spec", "policy-templates")
+		if err != nil {
+			return nil, fmt.Errorf("invalid input Policy: %w", err)
+		}
+
+		if !found {
+			return nil, fmt.Errorf("invalid input Policy: no policy-templates found")
+		}
+
+		cfgPolFound := false
+
+		for i, tmpl := range tmpls {
+			tmplMap, ok := tmpl.(map[string]any)
+			if !ok {
+				continue
+			}
+
+			objDef, ok := tmplMap["objectDefinition"]
+			if !ok {
+				continue
+			}
+
+			objDefMap, ok := objDef.(map[string]any)
+			if !ok {
+				continue
+			}
+
+			if objDefMap["kind"] != "ConfigurationPolicy" {
+				continue
+			}
+
+			if cfgPolFound {
+				cmd.Println("Ignoring additional ConfigurationPolicy in input policy")
+
+				continue
+			}
+
+			cfgPolFound = true
+
+			cfgpolBytes, err := json.Marshal(objDef)
+			if err != nil {
+				return nil, fmt.Errorf("unable to marshal policy template back to JSON")
+			}
+
+			if err := k8syaml.UnmarshalStrict(cfgpolBytes, &cfgpol); err != nil {
+				return nil, fmt.Errorf("could not unmarshal input policy template [%v] to a "+
+					"ConfigurationPolicy: %w", i, err)
+			}
+		}
+
+		if !cfgPolFound {
+			return nil, fmt.Errorf("invalid input Policy: it must contain a ConfigurationPolicy")
+		}
+	default:
+		return nil, fmt.Errorf("unsupported input kind: %v, must be 'Policy' or 'ConfigurationPolicy'",
+			unstruct.GetKind())
+	}
+
+	cfgpol.Spec.RemediationAction = policyv1.Inform
+	cfgpol.Spec.EvaluationInterval = policyv1.EvaluationInterval{
+		Compliant:    "10s",
+		NonCompliant: "10s",
+	}
+
+	cfgpol.OwnerReferences = []metav1.OwnerReference{{
+		Name: parentName,
+	}}
+
+	return &cfgpol, nil
+}
+
+// readInputResources takes stdin and any paths given as "positional" arguments,
+// and decodes them from YAML into k8s resources. Directories can be passed in
+// arguments: in this case all files in that directory will be read and decoded
+// (but no "deeper" directories are considered).
+func (d *DryRunner) readInputResources(cmd *cobra.Command, args []string) (
```

Comment:
Not sure if this is useful, but this is how I pull up all nested YAML files from the current directory (could be modified for your case):

```go
func getYAMLFiles() ([]string, error) {
	workingDir, err := os.Getwd()
	if err != nil {
		return nil, err
	}

	yamlFiles := []string{}

	err = filepath.WalkDir(workingDir, func(path string, d fs.DirEntry, err error) error {
		if err != nil {
			return nil
		}

		if d.IsDir() {
			return nil
		}

		fileExt := filepath.Ext(d.Name())
		if fileExt == ".yaml" || fileExt == ".yml" {
			yamlFiles = append(yamlFiles, path)
		}

		return nil
	})
	if err != nil {
		return nil, err
	}

	return yamlFiles, nil
}
```